planner/core/indexmerge_path.go

```diff
@@ -526,34 +601,42 @@ func (ds *DataSource) generateIndexMerge4MVIndex(normalPathCnt int, filters []ex
 			continue
 		}
 
-		partialPaths, isIntersection, err := ds.buildPartialPaths4MVIndex(accessFilters, idxCols, ds.possibleAccessPaths[idx].Index)
+		partialPaths, isIntersection, ok, err := ds.buildPartialPaths4MVIndex(accessFilters, idxCols, ds.possibleAccessPaths[idx].Index)
 		if err != nil {
 			return nil, err
 		}
-
-		indexMergePath := &util.AccessPath{PartialIndexPaths: partialPaths}
-		indexMergePath.IndexMergeIsIntersection = isIntersection
-		indexMergePath.TableFilters = remainingFilters
-
-		// TODO: use a naive estimation strategy here now for simplicity, make it more accurate.
-		minEstRows, maxEstRows := math.MaxFloat64, -1.0
-		for _, p := range indexMergePath.PartialIndexPaths {
-			minEstRows = math.Min(minEstRows, p.CountAfterAccess)
-			maxEstRows = math.Max(maxEstRows, p.CountAfterAccess)
-		}
-		if indexMergePath.IndexMergeIsIntersection {
-			indexMergePath.CountAfterAccess = minEstRows
-		} else {
-			indexMergePath.CountAfterAccess = maxEstRows
+		if !ok {
+			continue
 		}
 
-		mvIndexPaths = append(mvIndexPaths, indexMergePath)
+		mvIndexPaths = append(mvIndexPaths, ds.buildPartialPathUp4MVIndex(partialPaths, isIntersection, remainingFilters))
 	}
 	return
 }
 
+// buildPartialPathUp4MVIndex builds these partial paths up to a complete index merge path.
+func (ds *DataSource) buildPartialPathUp4MVIndex(partialPaths []*util.AccessPath, isIntersection bool, remainingFilters []expression.Expression) *util.AccessPath {
+	indexMergePath := &util.AccessPath{PartialIndexPaths: partialPaths}
+	indexMergePath.IndexMergeIsIntersection = isIntersection
+	indexMergePath.TableFilters = remainingFilters
+
+	// TODO: use a naive estimation strategy here now for simplicity, make it more accurate.
+	minEstRows, maxEstRows := math.MaxFloat64, -1.0
+	for _, p := range indexMergePath.PartialIndexPaths {
+		minEstRows = math.Min(minEstRows, p.CountAfterAccess)
+		maxEstRows = math.Max(maxEstRows, p.CountAfterAccess)
+	}
+	if indexMergePath.IndexMergeIsIntersection {
+		indexMergePath.CountAfterAccess = minEstRows
+	} else {
+		indexMergePath.CountAfterAccess = maxEstRows
+	}
+	return indexMergePath
+}
+
 func (ds *DataSource) buildPartialPaths4MVIndex(accessFilters []expression.Expression,
-	idxCols []*expression.Column, mvIndex *model.IndexInfo) ([]*util.AccessPath, bool, error) {
+	idxCols []*expression.Column, mvIndex *model.IndexInfo) (
+	partialPaths []*util.AccessPath, isIntersection bool, ok bool, err error) {
```

Comment:
plz add comment for the `ok` stands for, especially the difference between `err`.